pkg/minishift/systemtray/systemtray.go

```diff
@@ -118,57 +118,53 @@ func getStatus(profileName string) int {
 	return DOES_NOT_EXIST
 }
 
-// Add newly created profiles to the tray
-func addNewProfilesToTray() {
-	for {
-		time.Sleep(40 * time.Second)
-
-		newProfilesList := profile.GetProfileList()
-		for _, profile := range newProfilesList {
-			submenusLock.Lock()
-			if _, ok := submenus[profile]; ok {
-				submenusLock.Unlock()
-				continue
-			} else {
-				submenu := systray.AddSubMenu(strings.Title(profile))
-				submenus[profile] = submenu
-				submenusLock.Unlock()
-				startMenu := submenu.AddSubMenuItem(START, "", 0)
-				stopMenu := submenu.AddSubMenuItem(STOP, "", 0)
-				submenusToMenuItemsLock.Lock()
-				ma := MenuAction{start: startMenu, stop: stopMenu}
-				submenusToMenuItems[profile] = ma
-				submenusToMenuItemsLock.Unlock()
-
-				go startStopHandler(icon.Running, profile, ma.start, START_PROFILE)
-
-				go startStopHandler(icon.Stopped, profile, ma.stop, STOP_PROFILE)
-			}
-		}
+// Add newly created profiles and remove deleted profiles from tray
+func updateTrayMenu() {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		fmt.Println("Failed to create watcher:", err)
+	}
+	profilesBaseDir := filepath.Join(constants.GetMinishiftHomeDir(), "profiles")
```

Comment:
Reconstructing the profile directory path here is not ideal IMO but there are places we have done this. Also this code is going for maintenance so I am kind of ok with it.

Reply:
@LalatenduMohanty I can add it to the constants and also change it in other places wherever needed, if it is ok to do it in this PR?

Reply:
@anjannath yes, that should be ok. But ideally it should not be a constant but a function which can the environment variable for Minishift home in to account.